### PR TITLE
Add support for GitHub-style alert

### PR DESCRIPTION
**Pull request by guoPhineas**
WIP
- Previous PR: #156

### CODE REVIEW

**Comment by guoPhineas** [comment]
https://github.com/user-attachments/assets/9df397e1-f4b9-4d38-849d-cc9a338074f2


Hello,

This PR is ready for review.

I added `.markdownQuoteAlertEnabled()` to enable this style:

image


Thanks.🤝

**Comment by ajram23** [comment]
Love it. Hope this gets added!

**Comment by LiYanan2004** [comment]
Sorry for the delay. Reviewing now!

**Comment by LiYanan2004** [comment]
@guoPhineas Two issues:
- `MarkdownText` does not support this new view modifier (I fixed that)
- There are some edge cases for GitHub-style alert parsing (added some test cases for you) Please fix and make sure they are all passed)

File: Tests/MarkdownViewTests/MarkdownQuoteAlertTests.swift [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+import Foundation
+import Markdown
+import SwiftUI
+import Testing
+
+@testable import MarkdownView
+
+@Suite("Markdown Quote Alert")
+struct MarkdownQuoteAlertTests {
+    @Test(
```
Comment (LiYanan2004) [inline Tests/MarkdownViewTests/MarkdownQuoteAlertTests.swift:15]: This test fails

File: Tests/MarkdownViewTests/MarkdownQuoteAlertTests.swift [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        
+        let renderedBody = bodyChildren
+            .map(MarkdownViewRenderer.plainText(of:))
+            .joined(separator: "\n")
+        
+        #expect(alert.title == fixture.expectedTitle)
+        #expect(renderedBody.contains(fixture.expectedFirstBodyParagraph))
+    }
+    
+    @Test(
```
Comment (LiYanan2004) [inline Tests/MarkdownViewTests/MarkdownQuoteAlertTests.swift:96]: this test fails

**Comment by guoPhineas** [comment]
@LiYanan2004 
Never mind for delay. Thanks for review.

I just fixed the issues you raised. But the test isn't also passed. I speculate that there may be code logic issues with the current testing:

The `.multipleParagraphs` test appears to have an issue in its body-extraction logic.

For this fixture, Swift Markdown produces the following AST:

- First `Paragraph`: `[!NOTE]` followed by `First paragraph.`
- Second `Paragraph`: `Second paragraph.`

However, when there is more than one child, the test uses:

```swift
bodyChildren = Array(children.dropFirst())
```

This discards the first paragraph, including `First paragraph.`, and then asserts that the rendered body contains `First paragraph.`. Those two operations conflict.

The renderer should strip the alert marker/title line from the first paragraph, preserve the remaining inline content, and then append the remaining blockquote children. The implementation now follows that behavior, but the test’s extraction logic would need the same adjustment.

So, I fixed the issue in `commit cf6e19a`, how about this? It was passed on my Mac.

**Comment by LiYanan2004** [comment]
@guoPhineas  Here are some edge cases that need to be fixed.

image

**Comment by guoPhineas** [comment]
@LiYanan2004 

Thank you for your patient review and corrections.

I have just fixed these issues, and all tests have passed.

To test as many edge cases as possible, I used the Markdown text below for testing, and it displayed correctly.


https://github.com/user-attachments/assets/daa8ebaf-379e-4d28-b923-2fc657565398



```markdown
    # GitHub Alert Test Cases

    ## Valid Alerts

    > [!NOTE]
    > This is a standard note alert.

    > [!TIP]
    > This is a standard tip alert.

    > [!IMPORTANT]
    > This is a standard important alert.

    > [!WARNING]
    > This is a standard warning alert.

    > [!CAUTION]
    > This is a standard caution alert.

    ## Case-Insensitive Markers

    > [!note]
    > This lowercase marker should render as a note.

    > [!Tip]
    > This mixed-case marker should render as a tip.

    > [!important]
    > This lowercase marker should render as important.

    > [!Warning]
    > This mixed-case marker should render as a warning.

    > [!Caution]
    > This mixed-case marker should render as a caution.

    ## Body Layouts

    > [!NOTE]
    > The body starts immediately after the marker.
    > This line belongs to the same paragraph.

    > [!TIP]
    >
    > The body starts after a blank quoted line.

    > [!IMPORTANT]
    > First paragraph.
    >
    > Second paragraph.
    >
    > Third paragraph.

    > [!CAUTION]
    > A body containing **bold text**, *italic text*, `inline code`,
    > [a link](https://github.com), and ~~strikethrough text~~.

    ## Nested Block Content

    > [!NOTE]
    > - First list item
    > - Second list item
    >   - Nested list item

    > [!TIP]
    > 1. First ordered item
    > 2. Second ordered item

    > [!IMPORTANT]
    > > A nested blockquote.
    > >
    > > Another nested blockquote paragraph.

    > [!WARNING]
    > ```swift
    > let warning = "Code block inside an alert"
    > print(warning)
    > ```

    > [!CAUTION]
    > | Column A | Column B |
    > |----------|----------|
    > | Value A  | Value B  |

    ## Marker-Only Alerts

    > [!NOTE]

    > [!TIP]

    > [!IMPORTANT]

    > [!WARNING]

    > [!CAUTION]

    ## Surrounding Whitespace

    >    [!NOTE]   
    > The marker has surrounding whitespace.

    > [!WARNING]   
    > The marker has trailing spaces before the line break.
```

Thank you.

**Comment by guoPhineas** [comment]
Additionally, regarding the following cases, I have looked into them and it seems that they should not be supported. If I am mistaken, please feel free to correct me.

```markdown
> `[!NOTE]` is the token used to create a GitHub alert.
> Place it on a dedicated line.


> **[!WARNING]** is an example marker.
> This paragraph documents the syntax.
```

**Comment by ajram23** [comment]
@guoPhineas curious why? please check bear notes, it does support both.

**Comment by guoPhineas** [comment]
@ajram23 There are no universal guidelines ([GFM](https://github.github.com/gfm/) or [CommonMark](https://spec.commonmark.org/dingus/?text=%3E%20%60%5B!NOTE%5D%60%20is%20the%20token%20used%20to%20create%20a%20GitHub%20alert.%0A%3E%20Place%20it%20on%20a%20dedicated%20line.%0A%0A%0A%3E%20**%5B!WARNING%5D**%20is%20an%20example%20marker.%0A%3E%20This%20paragraph%20documents%20the%20syntax.)) for this quote style. So, I based it on the style of GitHub and Typora (v1.9.3).

On GitHub, the syntax looks like:
> `[!NOTE]` is the token used to create a GitHub alert.
> Place it on a dedicated line.


> **[!WARNING]** is an example marker.
> This paragraph documents the syntax.
---
Also, I referenced some discussions in the community:

- https://github.com/orgs/community/discussions/16925
- https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
---
Should we support this? @LiYanan2004
